fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -195,30 +200,22 @@ def _pmf_predict(self, X):
     def _format_results(self, gaps, Qs, lagrangian, B, eta_min):
         gaps_series = pd.Series(gaps)
         gaps_best = gaps_series[gaps_series <= gaps_series.min() + _PRECISION]
-        best_t = gaps_best.index[-1]
-        weights = Qs[best_t]
+        self._best_t = gaps_best.index[-1]
+        self._weights = Qs[self._best_t]
         hs = lagrangian.hs
         for h_idx in hs.index:
-            if h_idx not in weights.index:
-                weights.at[h_idx] = 0.0
+            if h_idx not in self._weights.index:
+                self._weights.at[h_idx] = 0.0
 
-        def best_classifier(X): return _mean_pred(X, hs, weights)
-        best_gap = gaps[best_t]
-
-        last_t = len(Qs) - 1
-
-        result = ExponentiatedGradientResult(
-            best_classifier,
-            best_gap,
-            lagrangian,
-            weights,
-            last_t,
-            best_t)
+        self._last_t = len(Qs) - 1
+        self._best_classifier = lambda X: _mean_pred(X, hs, self._weights)
+        self._best_gap = gaps[self._best_t]
+        self._classifiers = lagrangian.classifiers
+        self._n_oracle_calls = lagrangian.n_oracle_calls
```

Comment:
If these are supposed to be public, shouldn't they have a trailing underscore and not a leading one, to match sklearn conventions?

Reply:
It's not 100% clear which ones are meant to be public right now. There's another work item to match naming conventions. I stuck to what's been done so far. All of the former result object contents could be public in my opinion, but maybe @MiroDudik  has an opinion either way.

Reply:
Thought about this a little more. Since most of these are subject to be renamed (see my proposal for an idea about what I mean https://github.com/fairlearn/fairlearn-proposals/issues/4) I think it's wise to keep them private until that's happened. We don't want people to think that those attributes are there to stay (if we do a release), just to rename them in the immediately following release. 
I am working through all the refactoring items suggested by @MiroDudik so I will get to #256 as well eventually. This directly targets such things, just in case I miss out on it as part of the reductions refactoring. Does that make sense?